src/screens/Calculator/Calculator.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { PureComponent } from 'react';
+import CalculatorKey, { keyTypes } from './CalculatorKey/CalculatorKey';
+import DisplayComponent from '../../shared/components/Display/Display';
+import './styles.scss';
+
+class CalculatorComponent extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: '',
+    };
+  };
+
+  clearDisplay = () => {
+    this.setState({ display: '' });
+  }
+
+  isOperator = (key) => {
+    const operators = ['+', '-', '*', '/'];
+    return operators.includes(key);
+  }
+
+  checkLastElementOnDisplay = () => {
+    const lastElem = this.state.display.slice(-1);
+    return this.isOperator(lastElem);
+  };
+
+  addNewElement = val => {
+    this.setState(prevState => ({
+      display: prevState.display + val,
+    }))
+  };
+
+  replaceLastElement = val => {
+    const display = this.state.display;
+    const displayReplaced = display.slice(0, -1) + val;
+    this.setState({
+      display: displayReplaced,
+    })
+  }
+
+  addToDisplay = (val) => {
+    // this.checkLastElementOnDisplay() ? this.replaceLastElement(val) : this.addNewElement(val)
+    if (this.isOperator(val)) {
+      this.checkLastElementOnDisplay(val) ? this.replaceLastElement(val) : this.addNewElement(val)
+    } else {
+        this.addNewElement(val)
+    }
+  }
+
+    handleEqual = () => {
+    const display = this.state.display;
+    const lastElem = display.slice(-1);
+    const operators = ['+', '-', '*', '/'];
+    if (operators.includes(lastElem) || display.length === 1) {
```

Comment:
You're duplicating code here. You have separate method that does that - use it.

Reply:
Done!